packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -210,14 +210,15 @@ class LocalEngineArtifacts extends Artifacts {
     return fs.path.basename(engineOutPath);
   }
 
-  String _genSnapshotPath(TargetPlatform platform, BuildMode mode) {
-    String clang;
-    if (platform == TargetPlatform.ios) {
-      clang = 'clang_x64';
-    } else {
-      clang = getCurrentHostPlatform() == HostPlatform.darwin_x64 ? 'clang_i386' : 'clang_x86';
+  String _genSnapshotPath() {
+    const List<String> clangDirs = const <String>['clang_x86', 'clang_x64', 'clang_i386'];
+    final String genSnapshotName = _artifactToFileName(Artifact.genSnapshot);
+    for (String clangDir in clangDirs) {
+      final String genSnapshotPath = fs.path.join(engineOutPath, clangDir, genSnapshotName);
+      if (fs.file(genSnapshotPath).existsSync())
+        return genSnapshotPath;
     }
-    return fs.path.join(engineOutPath, clang, _artifactToFileName(Artifact.genSnapshot));
+    throw new Exception('Unable to find $genSnapshotName');
```

Comment:
Yes - this is an exception that generates a "Flutter crash report" log.

This function runs when you are building a debug mode JIT snapshot with a locally built engine.  If it throws, it means that something is wrong with your local engine build output.  It may or may not be a bug in flutter_tools or in the engine build process.